### ISSUE TITLE
Install script fails with cache-busting URL parameter

### ISSUE THREAD

**Issue by jrimmer** [Dicklesworthstone/beads_rust#8]
## Description

The suggested install command with cache-busting parameter fails:

```bash
curl -fsSL "https://raw.githubusercontent.com/Dicklesworthstone/beads_rust/main/install.sh?$(date +%s)" | bash
```

**Error:**
```
bash: line 420: v: command not found
```

## Workaround

The simpler command without the cache-busting parameter works fine:

```bash
curl -sSL "https://raw.githubusercontent.com/Dicklesworthstone/beads_rust/main/install.sh" | bash
```

## Environment

- **macOS:** 26.3 (Build 25D5087f)
- **Claude Code:** 2.1.12
- **Shell:** bash

## Notes

Line 420 in the current script is `check_gum || true`, which shouldn't cause this error. The issue may be related to:
- Cached older version of the script
- Interaction between the URL query parameter and how the script is processed
- Environment variable interference (e.g., `INSTALLER_VERSION` being set externally)

The suggested workaround in the README should perhaps omit the `?$(date +%s)` cache-busting suffix, or the root cause should be investigated further.

**Comment by tmchow** [comment]
Thanks for this workaround @jrimmer. I just hit this.

**Comment by Dicklesworthstone** [comment]
Thanks for the detailed report. I've verified this issue.

The cache-busting parameter itself isn't the problem - it's harmless in the URL. The "v: command not found" error on line 420 is strange because line 420 (`check_gum || true`) shouldn't produce that.

My hypothesis: you may have had an older cached version of the script that was different from what's currently at line 420. The installer has gone through several iterations.

**Recommendation**: The simpler command without the cache-busting suffix is just as good in practice:
```bash
curl -fsSL https://raw.githubusercontent.com/Dicklesworthstone/beads_rust/main/install.sh | bash
```

GitHub's CDN cache is short-lived anyway, so the cache-busting trick isn't usually necessary. I'll update the README to use the simpler form.

If you encounter this again with the current script version, please share the exact error output and I'll investigate further.

**Comment by jrimmer** [comment]
> Thanks for the detailed report. I've verified this issue.
> 
> The cache-busting parameter itself isn't the problem - it's harmless in the URL. The "v: command not found" error on line 420 is strange because line 420 (`check_gum || true`) shouldn't produce that.
> 
> My hypothesis: you may have had an older cached version of the script that was different from what's currently at line 420. The installer has gone through several iterations.
> 
> **Recommendation**: The simpler command without the cache-busting suffix is just as good in practice:
> 
> curl -fsSL https://raw.githubusercontent.com/Dicklesworthstone/beads_rust/main/install.sh | bash
> GitHub's CDN cache is short-lived anyway, so the cache-busting trick isn't usually necessary. I'll update the README to use the simpler form.
> 
> If you encounter this again with the current script version, please share the exact error output and I'll investigate further.

Yeah, sorry "Dickie ol' boy". I should've checked the issue title and conclusion as the error has nothing to do with cache-busting. The script "error" is real, though as I hadn't downloaded or otherwise used it prior to this attempt. I apologize for the ambiguity but it's the removal of the '-f' with curl that I believe actually made the difference.

**Comment by Dicklesworthstone** [comment]
Thanks for the clarification\! So the key difference was:

```bash
# This failed:
curl -fsSL "https://..." | bash

# This worked:
curl -sSL "https://..." | bash
```

The `-f` (fail silently) flag should cause curl to exit non-zero on HTTP errors rather than affecting bash parsing. That's odd behavior.

Could you try the current install command and report back?
```bash
curl -fsSL https://raw.githubusercontent.com/Dicklesworthstone/beads_rust/main/install.sh | bash
```

If it still fails, the full error output would help. The "v: command not found" error suggests something in the script is being parsed incorrectly—possibly a shell compatibility issue or download truncation.

For now, the working command without `-f` is fine as a workaround.

**Comment by jrimmer** [comment]
I ran into the same error on a fresh machine: (`bash: line 443: v: command not found`). Here's CC's take:

**Root Cause:** Line 297 has a quoting issue:

```bash
"$(gum style --foreground 42 '🔗 br installer v'${INSTALLER_VERSION})" \
```

The `v` before `${INSTALLER_VERSION}` is outside the single quotes but the variable isn't properly quoted. This can cause word splitting where `v` is interpreted as a command when the shell parses this in certain contexts.

**Fix:** Change line 297 to use consistent quoting:

```bash
"$(gum style --foreground 42 "🔗 br installer v${INSTALLER_VERSION}")" \
```

Or with proper concatenation:

```bash
"$(gum style --foreground 42 '🔗 br installer v'"${INSTALLER_VERSION}")" \
```

The line number discrepancy (420 vs 443) is likely due to different script versions, but the underlying bug is the same quoting issue on the `usage()` function's gum style call.

**Comment by Dicklesworthstone** [comment]
Fixed in 73cd0bb alongside #21. The root cause was unguarded access to `BASH_SOURCE[0]` when the script runs with `set -u` and is piped. The fix ensures piped execution (`curl ... | bash`) works correctly by providing a default value and explicitly checking for empty BASH_SOURCE.
